src/vs/workbench/parts/extensions/electron-browser/extensionTipsService.ts

```diff
@@ -68,16 +68,20 @@ export class ExtensionTipsService implements IExtensionTipsService {
 		if (!this.contextService.hasWorkspace()) {
 			return TPromise.as([]);
 		}
-		return this.fileService.resolveContent(this.contextService.toResource(paths.join('.vscode', 'extensions.json'))).then(content => { //TODO@Sandeep (https://github.com/Microsoft/vscode/issues/29242)
-			const extensionsContent = <IExtensionsContent>json.parse(content.value, []);
-			if (extensionsContent.recommendations) {
-				const regEx = new RegExp(EXTENSION_IDENTIFIER_PATTERN);
-				return extensionsContent.recommendations.filter((element, position) => {
-					return extensionsContent.recommendations.indexOf(element) === position && regEx.test(element);
-				});
-			}
-			return [];
-		}, err => []);
+		return TPromise.join(
+			this.contextService.getWorkspace().roots.map(root =>
+				this.fileService.resolveContent(root.with({ path: root.path + '/.vscode/extensions.json' })).then(content => {
```

Comment:
>That would use backslashes on Windows

True, but it is a valid file path. Creating an URI using `URI.file` takes care of it
Good thing is it will hide the knowledge of path separators.